### PR TITLE
Add the ability of ESO to reconcile only in scoped namespace

### PR DESCRIPTION
**Pull request by spirosoik**
By default ESO reconciles `Kind=ExternalSecret` across every namespace
in a k8s cluster. With the new flag `--nameespace` we can scope the
reconciling only to the provided namespace.

Ticket: #289

### CODE REVIEW

**Comment by spirosoik** [comment]
@moolen have a look

**Comment by knelasevero** [comment]
this LGTM! Looking at the AC of the issue it seems done also. Gonna mark it as approved and wait if moolen can also have a look. If he is unavailable until next week I merge this on Monday or Tuesday.

**Comment by moolen** [comment]
Hey, sorry i was sick the past 2 weeks :/
This looks good to me, thank you for picking this up! 

For the future we may want to have dedicated RBAC roles for a ESO when using `scopedNamespace` that restrict reading/updating secrets to a particular namespace. But it is totally fine for now to have one ClusterRole.

**Review by moolen — APPROVED** [review]

**Comment by knelasevero** [comment]
:thinking:

**Comment by kuzm1ch** [comment]
+1 to have dedicated RBAC role configuration for scoping ESO to specific namespaces

**Pull request merged**
